KenticoOnboardingApplication/test/KenticoOnboardingApplication.Api.Tests/Controllers/ListControllerTests.cs

```diff
@@ -0,0 +1,297 @@
+                    nameof(Item.LastUpdateTime))
+            };
+
+        [SetUp]
+        public void SetUp()
+        {
+            _repository = Substitute.For<IListRepository>();
+            _urlLocator = Substitute.For<IUrlLocator>();
+            _getItemService = Substitute.For<IGetItemService>();
+            _createItemService = Substitute.For<ICreateItemService>();
+            _updateItemService = Substitute.For<IUpdateItemService>();
+
+            _controller = new ListController(_repository, _urlLocator, _getItemService, _createItemService,
+                _updateItemService)
+            {
+                Configuration = new HttpConfiguration(),
+                Request = new HttpRequestMessage()
+            };
+        }
+
+        [Test]
+        public async Task GetAllItems_ReturnsItemsAndOk()
+        {
+            var expectedItems = new[]
+            {
+                ItemsCreator.CreateItem(id: s_ids[0], text: s_texts[0]),
+                ItemsCreator.CreateItem(id: s_ids[1], text: s_texts[1]),
+                ItemsCreator.CreateItem(id: s_ids[2], text: s_texts[2])
+            };
+
+            _repository.GetAllItemsAsync().Returns(Task.FromResult<IEnumerable<Item>>(expectedItems));
+
+            var (executedResult, items) =
+                await GetExecutedResultAndValue<IEnumerable<Item>>(controller => controller.GetAllItemsAsync());
+
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(items, Is.EqualTo(expectedItems).AsCollection);
+        }
+
+        [Test]
+        public async Task GetItem_WithExistingId_ReturnsItemAndOk()
+        {
+            var itemAndRetrievedItem = ItemsCreator.CreateItemAndRetrievedItem(id: s_ids[0], text: s_texts[0]);
+            _getItemService
+                .GetItemAsync(itemAndRetrievedItem.item.Id)
+                .Returns(itemAndRetrievedItem.retrievedItem);
+
+            var (executedResult, item) =
+                await GetExecutedResultAndValue<Item>(controller =>
+                    controller.GetItemAsync(itemAndRetrievedItem.item.Id));
+
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(item, Is.EqualTo(itemAndRetrievedItem.item));
+        }
+
+        [Test]
+        public async Task GetItem_WithEmptyId_ReturnsBadRequest()
+        {
+            var (executedResult, error) =
+                await GetExecutedResultAndValue<HttpError>(controller =>
+                    controller.GetItemAsync(Guid.Empty));
+            var modelStateKeys = error.ModelState.Keys;
+
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(modelStateKeys.Contains(nameof(Item.Id)), Is.True);
+            Assert.That(modelStateKeys.Count, Is.EqualTo(1));
+        }
+
+        [Test]
+        public async Task GetItem_WithNonexistingId_ReturnsNotFound()
```

Comment:
```suggestion
        public async Task GetItem_WithNonexistentId_ReturnsNotFound()
```
\- or -
```suggestion
        public async Task GetItem_WithNotExistingId_ReturnsNotFound()
```
(some spellchecker should really underline this for you…)